tests/Unit/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Test_TOV.cpp

```diff
@@ -0,0 +1,95 @@
+#include <memory>
+#include <pup.h>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/TOV.hpp"
+#include "PointwiseFunctions/EquationsOfState/EquationOfState.hpp"  // IWYU pragma: keep
+#include "Utilities/ConstantExpressions.hpp"
+
+// IWYU pragma: no_forward_declare EquationsOfState::EquationOfState
+
+namespace {
+
+constexpr double polytropic_constant = 8.0;
+double expected_newtonian_radius() noexcept {
+  /* Analytic solution in Newtonian limit for polytropic TOV integration has
+  $R = \pi / \alpha$ ; $M = 4\pi^{2}\rho_{0c} / \alpha^{3}$
+   Here $\rho_{0c}$ is the central mass density and $\alpha = sqrt(2\pi/K)$
+   with $K$ the polytropic constant
+  */
+  return M_PI / sqrt(2.0 * M_PI / polytropic_constant);
+}
+double expected_newtonian_mass(const double central_mass_density) noexcept {
+  return 4.0 * central_mass_density * square(M_PI) /
+         (cube(sqrt(2.0 * M_PI / polytropic_constant)));
+}
+void test_tov(const std::unique_ptr<EquationsOfState::EquationOfState<true, 1>>&
+                  equation_of_state,
+              const double central_mass_density,
+              const double final_log_enthalpy,
+              const bool newtonian_limit) noexcept {
+  Approx custom_approx = Approx::custom().epsilon(1.0e-08).scale(1.0);
+  const double surface_log_enthalpy = 0.000;
+  tov::InterpolationOutput tov_out_full(equation_of_state, central_mass_density,
+                                        surface_log_enthalpy);
+  tov::InterpolationOutput tov_out_test(equation_of_state, central_mass_density,
+                                        final_log_enthalpy);
+  if (newtonian_limit) {
+    const double final_radius{tov_out_full.outer_radius()};
+    const double final_mass{tov_out_full.mass(expected_newtonian_radius())};
+    CHECK(expected_newtonian_radius() == custom_approx(final_radius));
+    CHECK(expected_newtonian_mass(central_mass_density) ==
+          custom_approx(final_mass));
+  }
+  const double intermediate_radius{tov_out_test.outer_radius()};
+  const double intermediate_mass{tov_out_test.mass(intermediate_radius)};
+  const double intermediate_log_enthalpy{
+      tov_out_test.log_specific_enthalpy(intermediate_radius)};
+  const double interpolated_mass{tov_out_full.mass(intermediate_radius)};
+  const double interpolated_log_enthalpy{
+      tov_out_full.log_specific_enthalpy(intermediate_radius)};
+  CHECK(intermediate_mass == custom_approx(interpolated_mass));
+  CHECK(intermediate_log_enthalpy == custom_approx(interpolated_log_enthalpy));
+}
+
+SPECTRE_TEST_CASE(
+    "Unit.PointwiseFunctions.AnalyticSolutions.EinsteinSolutions.TOV",
+    "[Unit][PointwiseFunctions]") {
+  std::unique_ptr<EquationsOfState::EquationOfState<true, 1>>
+      equation_of_state =
+          std::make_unique<EquationsOfState::PolytropicFluid<true>>(
+              polytropic_constant, 2.0);
+
+  size_t num_pts = 25;
+  for (size_t i = 0; i < num_pts; i++) {
```

Comment:
It looks like each iteration of the loop is running the same code twice, except for a change in the values of `central_density` and `newtonian_limit` (argument to `test_tov`).

The code duplication could be removed by, for instance, using a lambda:
```
const auto wrap_test_tov = [&equation_of_state](
    const double central_density, const bool newtonian_limit) {
  // work to prepare test_tov args
 test_tov(...);
};
for (...) {
  wrap_test_tov(1.0e-10, true);
  wrap_test_tov(1.0e-3, false);
}
```

Alternatively, could `test_tov` be modified to include some of the "preparation" code, so that the loop can become:
```
for (...) {
  test_tov(..., 1.0e-10, true);
  test_tov(..., 1.0e-3, false);
}
```
?

Reply:
Hi Francois, 
I implemented the alternative option